src/dbnode/storage/database.go

```diff
@@ -378,7 +395,10 @@ func (d *db) newDatabaseNamespaceWithLock(
 			return nil, err
 		}
 	}
-	return newDatabaseNamespace(md, d.shardSet, retriever, d, d.commitLog, d.opts)
+	nsID := md.ID().String()
```

Comment:
Fine to alloc string here?

Reply:
Similarly - thankfully this only happens on changes. I figure given very low frequency of changes and the low number of namespaces (less than 10) it should be ok for this specific call-site.